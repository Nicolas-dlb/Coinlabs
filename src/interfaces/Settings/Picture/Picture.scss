.settings_picture {
  width: 86%;
  align-self: flex-end;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  padding: 20px;
  border-radius: 15px;
  background: var(--third-bg-color);
  margin-top: 2vh;
  transition: background-color 0.3s ease;
  padding-bottom: 25px !important;
  .settings_picture_container {
    display: flex;
    align-items: center;

    .settings_picture_message {
      margin-left: 10px;

      color: var(--third-font-color);
    }
  }
  h3 {
    margin-bottom: 3vh;
  }

  .set_profil_picture {
    background: #2b263f;
    border-radius: 5px;
    padding: 7px 10px;
    cursor: pointer;
    border: none;
    outline: none;
    color: var(--primary-font-color);
    font-family: inherit;
  }
}
