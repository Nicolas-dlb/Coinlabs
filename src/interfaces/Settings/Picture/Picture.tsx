/* eslint-disable no-unused-vars */
import React from "react";
import "./Picture.scss";
import { selectProfilPicture } from "utils/utils";

type PictureProps = {
  setPicture: (picture: string) => void;
};

function Picture({ setPicture }: PictureProps) {
  return (
    <div className="settings_picture">
      <h3>Profil picture</h3>
      <div className="settings_picture_container">
        <div className="settings_picture_input">
          <button
            type="button"
            tabIndex={0}
            onKeyDown={() => document.getElementById("picture-input")!.click()}
            onClick={() => document.getElementById("picture-input")!.click()}
            className="set_profil_picture"
          >
            Select file
          </button>
          <input
            id="picture-input"
            type="file"
            accept="image/*"
            onChange={(e) =>
              selectProfilPicture(e, setPicture, "set_picture_message")
            }
            style={{ display: "none" }}
          />
        </div>
        <div id="set_picture_message" className="settings_picture_message">
          <p>Any file selected</p>
        </div>
      </div>
    </div>
  );
}

export default Picture;
