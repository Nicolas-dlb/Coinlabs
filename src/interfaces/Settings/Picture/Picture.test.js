import React from "react";
import { render } from "enzyme";
import { act } from "react-dom/test-utils";
import Picture from "./Picture";

describe("Picture", () => {
  let PictureComponent;
  beforeEach(async () => {
    await act(async () => {
      PictureComponent = render(<Picture />);
    });
  });
  it("should render correctly", () => {
    expect(PictureComponent).toMatchSnapshot();
  });
  it("should render Profil picture title", () => {
    expect(PictureComponent.find("h3").text()).toBe("Profil picture");
  });

  it("should render an input", () => {
    expect(PictureComponent.find("input").length).toBe(1);
  });
  it("should render a button called Select file", () => {
    expect(PictureComponent.find("button").first().text()).toBe("Select file");
  });
});
