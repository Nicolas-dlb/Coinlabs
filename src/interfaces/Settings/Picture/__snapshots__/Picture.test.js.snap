// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture should render correctly 1`] = `
<div
  class="settings_picture"
>
  <h3>
    Profil picture
  </h3>
  <div
    class="settings_picture_container"
  >
    <div
      class="settings_picture_input"
    >
      <button
        class="set_profil_picture"
        tabindex="0"
        type="button"
      >
        Select file
      </button>
      <input
        accept="image/*"
        id="picture-input"
        style="display:none"
        type="file"
      />
    </div>
    <div
      class="settings_picture_message"
      id="set_picture_message"
    >
      <p>
        Any file selected
      </p>
    </div>
  </div>
</div>
`;
